spec/support/medium_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+require 'pry'
+
+RSpec.shared_examples "a medium" do
+  describe "model validations" do
+    describe ".validates" do
+      it "must have a title" do
+        expect(model.new(title: nil, ranking: 0)).to_not be_valid
+      end
+
+      it "must have a ranking" do
+        expect(model.new(title: "t", ranking: nil)).to_not be_valid
+      end
+
+      it "must have a ranking of zero on creation" do
+        expect(model.new(title: "t", ranking: 1)).to_not be_valid
+        expect(model.new(title: "t", ranking: 0)).to be_valid
+      end
+    end
+  end
```

Comment:
In response your PR message, I think it actually would be worthwhile to test the `by_ranking` method on your model classes. Even though the implementation right now is just calling an ActiveRecord method, the tests (and controller code and such) don't know that, so avoiding a test for it would be assuming too much knowledge about the implementation details of our model classes.

Here's something I might try for testing that method:
```ruby
describe "model scope" do
  describe ":by_ranking" do
    before :each do
      @model1 = model.create!(title: "test 1", ranking: 5)
      @model2 = model.create!(title: "test 2", ranking: 2)
      @model3 = model.create!(title: "test 3", ranking: 8)
    end
    let(:models) { model.all.by_ranking }
    it "returns all instances" do
      expect(models.length).to eq model.all.length
    end
    it "sorts in descending order by ranking column" do
      # Convert the `model.all.by_ranking` association into an array and compare it against our expected ordering
      expect(models.to_a).to eq [@model3, @model1, @model2]
    end
  end
end
```